src/analyzer.ts

```diff
@@ -15,8 +16,17 @@ export class Analyzer extends AnalyzerGen {
 
 	constructor(dartVMPath: string, analyzerPath: string) {
 		super();
+
 		console.log(`Starting Dart analysis server...`);
-		this.analyzerProcess = child_process.spawn(dartVMPath, [analyzerPath]);
+
+		let args = [analyzerPath];
+
+		// Optionally start the analyzer's diagnostic web server on the given port.
+		let port = getConfig('analyzerDiagnosticsPort');
+		if (port)
+			args.push(`--port=${port}`);
```

Comment:
That sounds really useful! Before I realised you were using 1-based lines/characters I was sending updates badly and getting really strange behaviour - being able to see the servers understanding of a file would've made that much easier =D

I'll have a nose next time I'm at PC!